[Miguel Tarcena] iP

proto-aiken-13 opened the pull request:
#**Introducing LatiBot!**

> "I've seen things, you people wouldn't believe, hmmm.
... attack ships on fire off the shoulder of Orion.
I've watched see Beams glitter in the dark near the Tannhauser Gate.
All those moments, will be lost in time like tears in rain..."
["... time to die ..."]

Do you feel lonely? Do you hate social interaction too? Do you want to have an unjudging, supportive friend who helps you remind of tasks that you need to do?

Well, too bad. Best I can give is LatiBot.

It's,
- Text-based.
- Easy to learn
- Reads like that one high school friend who's really into anime.
- ~~Fast~~ ~~Super Fast~~ **SLIGHTLY FASTER THAN SUPER FAST!**

Instructions:
1. Download it from [here](https://github.com/proto-aiken-13/ip/releases/tag/A-Jar).
2. Run it in a terminal.
3.  Add your tasks.
4.  Substitute your social life with it. 😉

Features
- [x] Manages tasks.
- [ ] Manages deadlines (coming soon!)
- [ ] Reminders (coming soon, too!)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```public class Main {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

proto-aiken-13 committed:
Level 0 of the iP
+ Added greeting and exit messages as a skeletal
programme

proto-aiken-13 committed:
Level 1 of the iP
+ Added a messageHandler method that merely echoes the user's input messages.

proto-aiken-13 committed:
Level 2 of the iP
+ Edited the messageHandler method to store all tasks inputted by the user accordingly.
+ Added a new DukeList class that stores and manages all messages given by the user.
+ The DukeList class contains the store and display methods which stores tasks inputted by the user and displays them in a list accordingly.

proto-aiken-13 committed:
Level 3 of the iP
+ Edited the messageHandler method in the Duke class to support marking and unmarking tasks.
+ Added a new Task class that stores a single task inputted by the user.
+ The DukeList class contains the mark and unmark methods which respectively marks and unmarks the current tasks  stored in the DukeList class.

proto-aiken-13 committed:
Level 3 of the iP
+ Edited the messageHandler method in the Duke class to support marking and unmarking tasks.
+ Added a new Task class that stores a single task inputted by the user.
+ The DukeList class contains the mark and unmark methods which respectively marks and unmarks the current tasks  stored in the DukeList class.

proto-aiken-13 committed:
+ Level - 4 Added Todo, Deadline and Event subclasses which inherit from Task.

+ Updated the messageHandler method in Duke to parse the commands given by the user to store specific tasks and their deadlines, if any.

+ Updated DukeList to overload its store methods, to accomodate to the different subclasses of Task.

proto-aiken-13 committed:
Added input and expected output tests

proto-aiken-13 committed:
+ Level 5
Modified Duke to make separate methods to greet and say bye to the user.

+ Changed the logic control flow of the messageHandler logic to use switch cases, as well as adding exceptions for invalid inputs.

proto-aiken-13 committed:
+ Level 6
+ Added a delete() method in DukeList to remove tasks and support to delete tasks in Duke.

proto-aiken-13 committed:
+ Level 6
+ Added a delete() method in DukeList to remove tasks and support to delete tasks in Duke.

proto-aiken-13 committed:
+ Level 6
+ Added a delete() method in DukeList to remove tasks and support to delete tasks in Duke.

proto-aiken-13 committed:
Add a new DataSaver class

The DataSaver class handles all saving and loading of tasks
in a test.s file.

The test.s file has the relative path of /data/data.s. This
makes it user-friendly across multiple platforms.

proto-aiken-13 committed:
Add a new Parser and DateAndTime classes

The new Parser class helps to parse commands given into the Duke class and also
helps to format date and times given to the Duke class.

This is done for OOP purposes, to help encapsulate tasks in managing messages received
by the Duke class.

proto-aiken-13 committed:
Add a new Parser and DateAndTime classes

The new Parser class helps to parse commands given into the Duke class and also
helps to format date and times given to the Duke class

This is done for OOP purposes, to help encapsulate tasks in managing messages received
by the Duke class

Remove unused import statements

proto-aiken-13 committed:
Add a new Parser and DateAndTime classes

The new Parser class helps to parse commands given into the Duke class and also
helps to format date and times given to the Duke class

This is done for OOP purposes, to help encapsulate tasks in managing messages received
by the Duke class

Remove unused import statements

Change task superclass into an abstract superclass

proto-aiken-13 committed:
Add a UI class.

The UI class contains multiple static methods to
display messages based on the inputs handled by
the DukeList class and the parser class.

This is to further abide to more OOP principles
to help encapsulate the roles of the overall
bot accordingly.

proto-aiken-13 committed:
Package all classes in the duke package

This is to easier encapsulate and package classes
for various aspects or behaviours of the Duke
bot.

proto-aiken-13 committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into A-Gradle

proto-aiken-13 committed:
Set up Gradle for testing

Modified the run path in the application entry point.

joel-foo commented:
Hi, I've been assigned to review your PR, but I think in your latest commit, you removed all your src files.

proto-aiken-13 committed:
Create a test directory to test ToDo and DateAndTime classes using JUnit

The ToDoTest class tests the toString and isDone methods, while the
DateAndTimeTest class tests the dayParser and isValidDate methods.

proto-aiken-13 committed:
Create a executable jar file

The executable jar file allows the application
to bedistributed easily to other users.

proto-aiken-13 committed:
Add a search method and support in DukeList, UI and parser classes

The search method helps users find specific tasks among their lists, for easier
management and tracking of their tasks.

proto-aiken-13 committed:
Ensure the code is done up to CS2103T coding standards

proto-aiken-13 committed:
Ensure the code is done up to CS2103T coding standards

proto-aiken-13 committed:
Update the DukeList, Parser and UI classes to support a search method.

This helps users to search any specific tasks in extremely long lists.

proto-aiken-13 committed:
Change JavaDocs to be up-to-standard with CS2103T JavaDocs
standards

proto-aiken-13 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/DukeList.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/UI.java

proto-aiken-13 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/UI.java

proto-aiken-13 committed:
Change naming and coding conventions to be up to standard with
CS2103T's coding standards

proto-aiken-13 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DukeList.java
#	src/main/java/duke/Parser.java

proto-aiken-13 committed:
Recommit some files (as they were not showing up for earlier commits)

proto-aiken-13 committed:
Recommit some files (as they were not showing up for earlier commits)

proto-aiken-13 committed:
Recommit some files (as they were not showing up for earlier commits)

proto-aiken-13 committed:
Recommit some files (as they were not showing up for earlier commits)

proto-aiken-13 committed:
Recommit some files (as they were not showing up for earlier commits)

proto-aiken-13 committed:
Change gradle tasks to include checkstyle support

Note: I have also attempted to install the checkstyle using the IDEA
method, but there were errors.

The checkstyle-idea.xml is found in the ip/.idea directory for your reference.

proto-aiken-13 committed:
Implement the checkstyle in the project

This is to help with future checkstyles as the scope of the project gets bigger.

andrefoo commented:
> Do note! Your `UI` class could be renamed to `Ui` to be in compliance with the module's Java coding standard. Abbreviations and acronyms should not be uppercase when used as a part of a name.

vivienherq commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,71 @@
+    /**The name of the task.*/
+    private String taskName;
+    /**Status on whether the task is done or not.*/
+    private boolean taskStatus;
```
Boolean variables should be named to sound like booleans.

tllshan replied:
Yep, so perhaps this could be renamed as "isDone", or "isCompleted"

proto-aiken-13 replied:
Thank you for your feedback! I missed that out...

vivienherq commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+/**
+ * The Duke class helps to manage all messages sent by the user to the chatbot accordingly.
+ */
+public class Duke {
```
LGTM!

tllshan commented on src/main/java/duke/DateAndTime.java:
```diff
@@ -0,0 +1,70 @@
+     * @param timing The time in HH:MM format
+     * @return The formatted date and time String.
+     */
+    public String dayParse(String date, String timing) {
```
If I'm not wrong, method names should be verbs? So perhaps this could be renamed as "parseDay" instead?

proto-aiken-13 replied:
True! Thanks for the feedback!

tllshan commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,147 @@
+     * @param deadline takes in the deadline of the deadline task.
+     */
+    public void store(String task, String deadline) {
+        Task todo = new Deadline(task, deadline);
```
I think it could get confusing if you name the Task as todo, when it's specifically a Deadline and not a ToDo. Perhaps you could consider renaming it to deadline? Also applies to the Event in the store method below (line 74).

tllshan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,173 @@
+     *
+     * @param tally takes in the input string.
+     */
+    public void messageHandler(String tally) {
```
I think method names should be verbs, instead of nouns? Maybe this could be renamed to "handleMessage"

tllshan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,173 @@
+        String comd = tally.split(" ")[0];
+
+        switch (comd) {
+            case "list":
```
If I'm not wrong, the coding standard for this course states that there should be no indentation for case clauses. So I think each case should be aligned with the switch?

proto-aiken-13 replied:
Hello!

Thanks for your feedback, unfortunately intelliJ keeps forcing to indent these switch-case things. Is there any way to stop it from doing so?

tllshan commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,157 @@
+     * Sends a greeting message to the user.
+     */
+    public static void greet() {
+        System.out.println("____________________________________________________________\n" +
```
The coding standard for this course states to break before an operator. So perhaps you could put the + on the next line, rather than at the end of the previous. I think doing this would make your code more readable! (Also applies to the rest of the methods in this class)

tllshan reviewed (commented):
Code looks generally well done to me, good job! Just a few small suggestions here and there to improve it further.

proto-aiken-13 committed:
Add a GUI for the DukeBot

This is to make it more
user-friendly and accessible.

proto-aiken-13 committed:
Add asserts to files

Unify variations of toSet() methods

There are several methods that finds the index of a task in the
TaskList. Thus, we need to check whether the given index is
valid.

Asserting the validity of said index ensures the correctness of
the index before finding the task itself.

Let's add asserts in the delete, mark and unmark methods in both
the TaskList and the Parser.

proto-aiken-13 committed:
DateAndTime: Update the parseDay method name (previously called
dayParse)

CS2103T coding conventions dictate that method names need to start
with verbs.

Let's update the dayParse method to abide to these standards.

proto-aiken-13 committed:
Ui: Remove the Scanner variable and import
DialogBox: Remove the flip method

To abide to coding standards, and with an especially large
number of files, all unused variables and imports must be
removed.

Removing such unused variables saves space and streamlines
the code as per the CS2103T coding standards, and indicates
improvement.

Let's remove the unused scanner methods in the Ui and the
flip methods in DialogBox, as there is no longer a need of
CLI scanners with the GUI and there is already a separate
class responsible for showing the opposing chatbot.

proto-aiken-13 committed:
Add support for the Duke bot to store trivia

Since DukeBot is essentially a task management
chatbot, it would also suffice to help it store
trivia given to it by the user for additional
help.

Let's add multiple files that can help it keep
track of data.

proto-aiken-13 committed:
Add support for the Duke bot to store trivia

Since DukeBot is essentially a task management
chatbot, it would also suffice to help it store
trivia given to it by the user for additional
help.

Let's add multiple files that can help it keep
track of data.

proto-aiken-13 committed:
Add a UG in the README.md file

To top things off for the iP, the user must, of course,
have a user guide to refer to.

Let's add a UG in the README.md.

Also other files have been updated to clear up bugs,
particularly caused by the Parser class in handling
task editing and trivia handling.

proto-aiken-13 committed:
Create Ui.png

Add a Ui.png in docs

proto-aiken-13 committed:
Merge pull request #3 from proto-aiken-13/branch-A-FinishingTouches

Branch a finishing touches

proto-aiken-13 committed:
Update Ui.png

Update the UI.png image

proto-aiken-13 committed:
Ui.png: Change image
Main.java: Add a header for the name of the Duke bot

proto-aiken-13 committed:
Merge pull request #4 from proto-aiken-13/branch-A-FinishingTouches

Update Ui.png

proto-aiken-13 committed:
Merge pull request #5 from proto-aiken-13/branch-A-FinishingTouches

Update Ui.png

proto-aiken-13 committed:
Update the README.md

proto-aiken-13 committed:
Merge pull request #6 from proto-aiken-13/branch-A-FinishingTouches

Update the README.md

proto-aiken-13 committed:
Update README.md

proto-aiken-13 committed:
Update the code to match CS2103T coding standards

A lot of the files still had some errors that I missed out on
earlier commits.

Let's iron out these issues!

proto-aiken-13 committed:
Parser: Fix a bug where deleting tasks returned a mark task.
DateFormatter: Refactor the name of the class into DateFormatter
(formerly DateAndTime)

A few more bugs persisted when smoke testing. In addition, the
instructed class to write was DateFormatter for formatting date
and time.

Let's make these changes accordingly!

proto-aiken-13 committed:
DeadlineTest: Amend the javadocs for the last test

proto-aiken-13 committed:
Merge pull request #9 from proto-aiken-13/branch-A-MoreCodeQuality

code quality

proto-aiken-13 committed:
Merge pull request #8 from proto-aiken-13/branch-A-Miscallaneous

branch miscallaneous
